src/cmd/deploy.ts

```diff
@@ -217,19 +238,25 @@ export default class DefenderDeploy {
   }
 
   private async deploySecrets(output: DeployOutput<string>) {
-    const secrets: YSecret[] = this.serverless.service.resources?.Resources?.secrets ?? [];
+    const globalSecrets: YSecret = this.serverless.service.resources?.Resources?.secrets?.global ?? {};
+    const stackSecrets: YSecret = this.serverless.service.resources?.Resources?.secrets?.stack ?? {};
+    const allSecrets = _.map(_.entries(Object.assign(globalSecrets, stackSecrets)), ([k, v]) => ({ [k]: v }));
+
     const client = getAutotaskClient(this.teamKey!);
     const retrieveExisting = () => client.listSecrets().then((r) => r.secretNames ?? []);
-
     await this.wrapper<YSecret, string>(
       this.serverless,
       'Secrets',
-      secrets,
+      allSecrets,
       retrieveExisting,
       // on update
-      async (secret: YSecret, match: any) => {
+      async (secret: YSecret, match: string) => {
+        const stackName = getStackName(this.serverless);
+        const [_, value] = Object.entries(secret)[0];
+        const stackSecretMatch = stackSecrets[match];
+        // if secret key matches that of any in stackSecrets, prepend the key with stackName
```

Comment:
What happens if I define two secrets with the same name (eg "foo"), one global and one stack? Wouldn't this consider them both as stack?

Reply:
Ha, yes :) Nice catch.

Reply:
Fixed in [this](https://github.com/OpenZeppelin/defender-serverless/pull/23/commits/f440449d8aec2296f6014fcc14ef10c4dc797715) commit.